modules/xmpp/ChatRoom.js

```diff
@@ -885,12 +896,12 @@ export default class ChatRoom extends Listenable {
                     .length) {
             this.discoRoomInfo();
         }
+        const jsonMessage = $(msg).find('>json-message').text();
+        const parsedJson = tryParseJSONAndVerify(jsonMessage);
 
-        const jsonPayload = tryParseJSONAndVerify(txt);
-
-        if (jsonPayload) {
+        if (parsedJson && stamp === undefined) {
```

Comment:
Please add a short comment which describes why the check for `stamp` is necessary